### PR TITLE
Add shell detection

### LINKED ISSUE #3: wrong shell
@ofek `userpath` doesn't work for my setup locally so I'm having some trouble testing.

Here are my dotfiles in case it helps: https://github.com/AlJohri/dotfiles

Issues:
- current shell is ZSH but it appends to bash files
- appends to both ~/.bashrc  and ~/.bash_profile instead of just doing the right thing based on the OS

```
$ userpath append /testinguserpath
Success!
$ echo "$?"
0
```

```
$ userpath verify /testinguserpath
The directory `/testinguserpath` is in PATH, pending a shell restart!

$ echo "$?"
2
```

```
$ echo "$SHELL"
/usr/local/bin/zsh
```

```
$ tail -n 2 ~/.bash_profile
# Created by `userpath` on 2019-05-10 15:39:32
export PATH="$PATH:/testinguserpath"
```

```
$ tail -n 2 ~/.bashrc
# Created by `userpath` on 2019-05-10 15:39:32
export PATH="$PATH:/testinguserpath"
```

`~/.zprofile` and `~/.zshrc` were untouched

### PR DESCRIPTION
**Pull request by ofek**
Resolves #3

### CODE REVIEW

**Comment by ofek** [comment]
@AlJohri Would you mind testing zsh?
@jaraco Would you mind testing xonsh?

I plan to add a Docker env for each shell to tests but I won't have time this week

**Comment by jaraco** [comment]
I'm getting an error when it runs, though it does seem to have added the expected lines:

```
userpath ofek/detect $ git-id                                                                                                                                                  
3cc42f8
userpath ofek/detect $ tox --notest                                                                                                                                            
GLOB sdist-make: /Users/jaraco/code/public/userpath/setup.py
python create: /Users/jaraco/code/public/userpath/.tox/python
python installdeps: pytest
python inst: /Users/jaraco/code/public/userpath/.tox/.tmp/package/1/userpath-1.1.0.zip
python installed: atomicwrites==1.3.0,attrs==19.1.0,Click==7.0,more-itertools==7.0.0,pluggy==0.11.0,py==1.8.0,pytest==4.5.0,six==1.12.0,userpath==1.1.0,wcwidth==0.1.7
___________________________________________________________________________________ summary ___________________________________________________________________________________
  python: skipped tests
  congratulations :)

userpath ofek/detect $ .tox/python/bin/python -m userpath append foo/bar                                                                                                       
An unexpected failure seems to have occurred.
userpath ofek/detect $ edit ~/.xonshrc                                                                                                                                         
userpath ofek/detect $ tail -n 3 ~/.xonshrc                                                                                                                                    

# Created by `userpath` on 2019-05-13 12:20:01
$PATH.append(r"/Users/jaraco/code/public/userpath/foo/bar")
```

File: userpath/shells.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def show_path_command(cls):
+        return ['fish', '--login', '-c', 'for p in $PATH; echo "$p"; end']
+
+
+class Xonsh(Shell):
+    def config(self, location, front=True):
+        locations = location.split(pathsep)
+
+        if front:
+            contents = '\n'.join('$PATH.insert(0, r"{}")'.format(location) for location in reversed(locations))
```
Comment (jaraco) [inline userpath/shells.py]: May I suggest `{!r}` instead of `r"{}"`?

Comment (ofek) [reply userpath/shells.py]: Much nicer, thanks!

**Comment by ofek** [comment]
@AlJohri @jaraco @cs01 So, here's what I've done:

- Implemented the ability to select particular shells to modify, defaulting to auto-detection, and if that fails, to `sh` and `bash`
- We now require https://github.com/nir0s/distro as a dependency to handle https://github.com/ofek/userpath/issues/3#issuecomment-492491977 and other issues like it in future
- Made integration tests via Docker for the login & non-login version of every shell, with full support for code coverage. Currently, we just test Debian (Stretch) but adding other Dockerfiles would be trivial

I'd appreciate it if you could test this once more, then I'll merge and release!

**Comment by jaraco** [comment]
Looking good!

```
userpath ofek/detect $ git-id
6ace5e6
userpath ofek/detect $ tox --notest -qq
userpath ofek/detect $ .tox/python/bin/python -m userpath append foo/bar
Success!
userpath ofek/detect $ tail -n 3 ~/.xonshrc

# Created by `userpath` on 2019-06-21 20:28:07
$PATH.append('/Users/jaraco/code/public/userpath/foo/bar')
```

Couple of comments follow.

Should there be output for success? Unix conventions say don't emit anything on success, although maybe that's an old convention. You may be asked later to add a `-q` operator to suppress that output. I don't feel strongly about it.

At first, the help text had me thinking that the Windows behavior was unnecessarily special cased. In particular:

> -a, --all-shells; Update PATH of all supported shells. This has no effect on Windows as that is the standard behavior.

This made me think that on Windows, it would attempt to modify the config of all these shells... then I realized that the reason "that is the standard behavior" means that environment settings always affect all shells, not that `userpath` will be modifying all supported shells. I'm not sure there's anything to do here, but I thought I'd share my momentary confusion.

Overall, no objections. Looks great to me.

**Comment by AlJohri** [comment]
I'll try and test this out early next week- apologies for the delay.

**Comment by ofek** [comment]
`1.2.0` is now on PyPI 😄 

@jaraco I incorporated your suggestions
@AlJohri Can we proceed with the Homebrew PR now?

**Pull request merged**